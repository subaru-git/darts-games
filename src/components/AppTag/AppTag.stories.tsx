import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import AppTag from './AppTag';

export default {
  title: 'Components/AppTag',
  component: AppTag,
} as ComponentMeta<typeof AppTag>;

const Template: ComponentStory<typeof AppTag> = (args) => <AppTag {...args} />;

export const Default = Template.bind({});
Default.args = {
  tag: 'cricket',
};
