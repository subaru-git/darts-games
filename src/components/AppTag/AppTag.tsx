import React, { FC } from 'react';
import { Tag } from '@chakra-ui/react';

type AppTagProps = {
  tag: string;
};

const AppTag: FC<AppTagProps> = ({ tag }) => {
  const color = ConstantTags.find((t) => t.name === tag)?.color || {
    font: 'gray.200',
    bg: 'gray.400',
  };
  return (
    <Tag color={color.font} backgroundColor={color.bg}>
      {tag}
    </Tag>
  );
};

const ConstantTags = [
  { name: 'BULL', color: { font: 'gray.100', bg: 'blue.400' } },
  { name: 'single', color: { font: 'gray.100', bg: 'cyan.600' } },
  { name: 'double', color: { font: 'gray.100', bg: 'cyan.400' } },
  { name: 'master', color: { font: 'gray.100', bg: 'cyan.800' } },
  { name: 'cricket', color: { font: 'gray.100', bg: 'teal.400' } },
  { name: '19', color: { font: 'gray.100', bg: 'green.400' } },
  { name: '20', color: { font: 'gray.100', bg: 'green.400' } },
  { name: 'count up', color: { font: 'gray.100', bg: 'pink.400' } },
  { name: 'finish', color: { font: 'gray.100', bg: 'orange.400' } },
  { name: 'simulation', color: { font: 'gray.100', bg: 'purple.400' } },
];

export default AppTag;
