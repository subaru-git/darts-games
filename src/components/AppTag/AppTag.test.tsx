import { render } from '@testing-library/react';
import AppTag from './AppTag';

test('should rendering', () => {
  const { container } = render(<AppTag tag='cricket' />);
  expect(container).toMatchSnapshot();
});
