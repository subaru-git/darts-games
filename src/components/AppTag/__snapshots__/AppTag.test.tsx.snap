// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <span
    class="css-1dl6jat"
  >
    cricket
  </span>
</div>
`;
